### PR TITLE
Add .csv .xlsx and .xlsx import for companies

### PR DESCRIPTION
**Pull request by mmarusyk**
[Trello Task](https://trello.com/https://trello.com/c/siFxPjfa/40-xls-import-for-companies)

**CHECK LIST**
- [x] specs added
- [x] specs passed
- [x] code coverage >90%
- [ ] DB indexes added if needed
- [ ] seed.rb file is updated if needed
- [x] PR is reviewed manually again (to make sure you have 100% ready code)
- [x] PR doesn't have conflicts with main branch
- [x] I've checked new feature as logged in and logged out user if needed
- [x] "let!" is used in specs
- [x] screenshot of feature is added if task has UI
fourth
third
second
first
Screenshot from 2020-10-28 12-14-47
Screenshot from 2020-10-28 12-13-44

Screenshot from 2020-11-06 20-45-51
Screenshot from 2020-11-06 20-47-22

### CODE REVIEW

File: app/importers/companies_import.rb [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  START_ROW = 2
+
+  attr_accessor :file
+
+  def initialize(attributes = {})
+    attributes.each { |name, value| send("#{name}=", value) }
+  end
+
+  def save
+    if find_errors(items).map { |c| c.errors.messages.empty? }.all?
```
Comment (pereslop) [inline app/importers/companies_import.rb:15]: select {|c| c.errors.messages.empty? }.empty? may be better

Comment (mmarusyk) [reply app/importers/companies_import.rb:15]: select return array of hashes and `[{},{}...].empty?` or `[{},{e: 'd'}...]` will be `false`.

**Review by matsevich — APPROVED** [review]

**Review by Ukhanskyi — APPROVED** [review]

**Review by marikrebega — APPROVED** [review]

File: app/controllers/account/admin/companies_imports_controller.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+      if @companies_import.save
+        redirect_to account_admin_companies_path, flash: { success: 'Companies were imported.' }
+      else
+        render :new
+      end
+    else
+      redirect_to new_account_admin_companies_import_path, flash: { danger: 'Field can not be empty!' }
+    end
+  end
+end
```
Comment (matsevich) [inline app/controllers/account/admin/companies_imports_controller.rb:18]: ти забув видалити цей контроллер?

**Review by Ukhanskyi — APPROVED** [review]

**Review by Andrii-Kozak — APPROVED** [review]

File: app/importers/applications_import.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  def initialize(attributes = {})
+    attributes.each { |name, value| send("#{name}=", value) } if !attributes.nil? && attributes.key?(:file)
+  end
+
+  def perform # rubocop:disable Metrics/MethodLength
+    if !file
+      errors.add :base, 'Field can\'t be empty!'
+      false
+    else
+      finding_errors
```
Comment (chu4) [inline app/importers/applications_import.rb:15]: again this check for errors should be redundant as it was for "Field can't be empty!"

File: spec/controllers/account/admin/companies_controller_spec.rb [hunk 4]
```diff
@@ -110,4 +111,32 @@
       expect(flash[:danger]).to be_present
     end
   end
+
```
Comment (chu4) [inline spec/controllers/account/admin/companies_controller_spec.rb:114]: cover #new_import and #import action test into their own describe block

and move let for file isede of it as there is no need for file in any other specs beside this two

File: app/importers/companies_import.rb [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    (2..open_spreadsheet.last_row).map do |i|
+      row = Hash[[open_spreadsheet.row(1), open_spreadsheet.row(i)].transpose]
+      company = Company.new(row_to_hash(row, %w[company_type email fax name phone website]))
+      company.build_address(row_to_hash(row, %w[city country state street]))
+      company.build_account(row_to_hash(row, %w[edrpou iban purpose]))
+      company
+    end
+  end
+
+  def validate_data
```
Comment (chu4) [inline app/importers/companies_import.rb:17]: can this be moved to parent class?

File: app/importers/applications_import.rb [hunk 6]
```diff
@@ -0,0 +1,39 @@
+class ApplicationsImport
+  include ActiveModel::Model
+
+  attr_accessor :file
+
+  def initialize(attributes = {})
+    attributes.each { |name, value| send("#{name}=", value) } if !attributes.nil? && attributes.key?(:file)
```
Comment (chu4) [inline app/importers/applications_import.rb:7]: attributes should never be nil as you have attributes = {}

File: app/importers/applications_import.rb [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    attributes.each { |name, value| send("#{name}=", value) } if attributes&.key?(:file)
+  end
+
+  def perform # rubocop:disable Metrics/MethodLength
+    if !file
+      errors.add :base, 'Field can\'t be empty!'
+      false
+    else
+      validate_data
+      finding_errors
```
Comment (chu4) [inline app/importers/applications_import.rb:16]: juset return false here no need for moving it to method

**Review by chu4 — APPROVED** [review]

**Pull request merged**